configuration/nfs.md

```diff
@@ -0,0 +1,304 @@
+
+An NFS is perhaps best suited to more permanent network-mounted directories, such as `/home` directories or regularly-accessed shared resources. If you want a network share that guest users can easily connect to, Samba is better suited to the task. This is because tools to temporarily mount and detach from Samba shares are more readily available across old and proprietary operating systems.
+
+Before deploying an NFS, you should be familiar with:
+
+* Linux file and directory permissions
+* mounting and unmounting filesystems
+
+## Set up a basic NFS server
+
+Install the packages required using the command below:
+
+```bash
+sudo apt install nfs-kernel-server
+```
+
+For easier maintenance, we will isolate all NFS exports in single directory, into which the real directories will be mounted with the `--bind` option.
+
+Suppose we want to export our users' home directories, which are in `/home/users`. First we create the export filesystem:
+
+```bash
+sudo mkdir -p /export/users
+```
+
+Note that `/export` and `/export/users` will need 777 permissions, as we will be accessing the NFS share from the client without LDAP/NIS authentication. This will not apply if using authentication (see below). Now mount the real `users` directory with:
+
+```bash
+sudo mount --bind /home/users /export/users
+```
+
+To save us from retyping this after every reboot, we add the following line to `/etc/fstab`:
+
+```
+/home/users    /export/users   none    bind  0  0
+```
+
+There are three configuration files that relate to an NFS server:
+
+1. `/etc/default/nfs-kernel-server`
+1. `/etc/default/nfs-common`
+1. `/etc/exports`
+
+The only important option in `/etc/default/nfs-kernel-server` for now is `NEED_SVCGSSD`. It is set to `"no"` by default, which is fine, because we are not activating NFSv4 security this time.
+
+In order for the ID names to be automatically mapped, the file `/etc/idmapd.conf` must exist on both the client and the server with the same contents and with the correct domain names. Furthermore, this file should have the following lines in the `Mapping` section:
+
+```
+[Mapping]
+
+Nobody-User = nobody
+Nobody-Group = nogroup
+```
+
+However, note that the client may have different requirements for the Nobody-User and Nobody-Group. For example, on RedHat variants, it is `nfsnobody` for both. If you're not sure, check via the following commands to see if `nobody` and `nogroup` are there:
+
+```bash
+cat /etc/passwd
+cat /etc/group
+```
+
+This way, server and client do not need the users to share same UID/GUID. For those who use LDAP-based authentication, add the following lines to the `idmapd.conf` of your clients:
+
+```
+[Translation]
+
+Method = nsswitch
+```
+
+This will cause `idmapd` to know to look at `nsswitch.conf` to determine where it should look for credential information. If you have LDAP authentication already working, `nsswitch` shouldn't require further explanation.
```

Comment:
Should this read "`nsswitch` shouldn't require further information" instead of "`nsswitch` shouldn't require further explanation"?

Reply:
I think what the author is getting at is that if the user has already set up LDAP authentication, then they should know about what nsswitch does since configuring nsswitch is required to enable it.